pkg/pipeline/extract/conntrack/store.go

```diff
@@ -73,57 +77,77 @@ func (cs *connectionStore) addConnection(hashId uint64, conn connection) {
 	cs.hashId2groupIdx[hashId] = groupIdx
 
 	groupLabel := cs.groups[groupIdx].labelValue
-	groupLen := cs.groups[groupIdx].mom.Len()
-	cs.metrics.connStoreLength.WithLabelValues(groupLabel).Set(float64(groupLen))
+	groupLen := cs.groups[groupIdx].runningMom.Len()
+	cs.metrics.runningConnStoreLength.WithLabelValues(groupLabel).Set(float64(groupLen))
 }
 
-func (cs *connectionStore) getConnection(hashId uint64) (connection, bool) {
+func (cs *connectionStore) getConnection(hashId uint64) (connection, bool, bool) {
 	groupIdx, found := cs.hashId2groupIdx[hashId]
 	if !found {
-		return nil, false
+		return nil, false, false
 	}
-	mom := cs.groups[groupIdx].mom
+	mom := cs.groups[groupIdx].runningMom
 
+	// get connection from running map
+	isRunning := true
 	record, ok := mom.GetRecord(utils.Key(hashId))
 	if !ok {
-		return nil, false
+		// fallback on expired map if not found
+		isRunning = false
+		mom := cs.groups[groupIdx].expiredMom
+		record, ok = mom.GetRecord(utils.Key(hashId))
+		if !ok {
+			return nil, false, false
+		}
```

Comment:
The unit test is available here: https://github.com/netobserv/flowlogs-pipeline/commit/1323e74b2dc4e354c9acedd04ddc97ff0a0fba56
It can be added to this PR.

Reply:
Awesome, thanks @ronensc !

I'm adding this to my next commit but feel free to directly contribute in my PRs when you want to add items like this :+1: